react-basic-practice/src/features/SignUpForm/index.jsx

```diff
@@ -1,86 +1,86 @@
   const [errorMessage, setErrorMessage] = useState(initialErrorMsgs);
   const [inputValue, setInputValue] = useState(initialInput);
+  const navigate = useNavigate();
+  const { loading, showLoading, hideLoading } = useLoading();
 
   const handleInputValue = (value) => {
     setInputValue({ ...inputValue, ...value });
   };
 
-  const validation = () => {
-    for (const key in inputValue) {
-      if (inputValue[key] === '') {
-        setErrorMessage((preMsg) => ({
-          ...preMsg,
-          [key]: 'This input is required.',
-        }));
-      } else {
-        setErrorMessage((preMsg) => ({
-          ...preMsg,
-          [key]: '',
-        }));
-      }
-    }
-  };
-
-  const validateInputs = () => {
-    if (inputValue.inputPassword !== '') {
-      REGEXP.REGEXP_PASSWORD.test(inputValue.inputPassword)
-        ? setErrorMessage((preMsg) => ({ ...preMsg, inputPassword: '' }))
-        : setErrorMessage((preMsg) => ({ ...preMsg, inputPassword: 'Invalid password' }));
-    }
-
-    if (inputValue.inputEmail !== '') {
-      REGEXP.REGEXP_MAIL.test(inputValue.inputEmail)
-        ? setErrorMessage((preMsg) => ({ ...preMsg, inputEmail: '' }))
-        : setErrorMessage((preMsg) => ({ ...preMsg, inputEmail: 'Invalid email' }));
-    }
-
-    if (inputValue.inputUserName !== '') {
-      REGEXP.REGEXP_USER_NAME.test(inputValue.inputUserName)
-        ? setErrorMessage((preMsg) => ({ ...preMsg, inputUserName: '' }))
-        : setErrorMessage((preMsg) => ({ ...preMsg, inputUserName: 'Invalid user name' }));
-    }
-  };
-
-  const handleSignUp = (e) => {
+  const handleSignUp = async (e) => {
     try {
       e.preventDefault();
 
-      setIsSignUpLoading(true);
-
-      validation();
-      validateInputs();
-
-      setIsSignUpLoading(false);
+      showLoading();
+
+      // Check validation input
+      const errorValid = validateInput(inputValue);
+      setErrorMessage(errorValid.validateError);
+
+      if (!errorValid.error) {
+        // Check email already exists.
+        const dataUser = await getAllUsers();
+        const haveUser = dataUser.some((user) => user.email === inputValue.email);
+        if (haveUser) {
+          // Show error if email already exists.
+          setErrorMessage((preMsg) => ({
+            ...preMsg,
+            form: 'Email is already in use. Please try another one.',
+          }));
+        } else {
+          // Send data to API to create new users.
+          const newUser = {
+            id: uuidv4(),
+            username: inputValue.userName || '',
+            email: inputValue.email || '',
+            password: inputValue.password || '',
+          };
+
+          const newCart = {
+            id: newUser.id,
+            listProducts: [],
+          };
+          await createUser(newUser);
+          await createNewCart(newCart);
+
+          navigate('/login');
+        }
+      } else {
+        e.target.reset();
```

Comment:
Please don't do this. You check `errorValid.error` if it's true then you do this `setErrorMessage(errorValid.validateError);`.